server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DomainMappingsRoutesTest.java

```diff
@@ -0,0 +1,217 @@
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import static com.jayway.restassured.RestAssured.*;
+import static org.apache.james.webadmin.WebAdminServer.NO_CONFIGURATION;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.entry;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.isEmptyString;
+import static org.mockito.Mockito.spy;
+
+class DomainMappingsRoutesTest {
+    private RecipientRewriteTable recipientRewriteTable;
+    private WebAdminServer webAdminServer;
+
+    @SuppressWarnings("unused")
+    static Stream<Arguments> invalidInputs() {
+        return Stream.of(
+                Arguments.of("domain.com", ""),
+                // Why this params should pass the test ?
+                // "domain.com, '    '",
+                // "domain.com, '    \n\t\r'",
+                Arguments.of("abc@domain.com", "domain.com"),
+                Arguments.of("domain.com", "abc@domain.com")
+
+        );
+    }
+
+    private void createServer(DomainMappingsRoutes domainMappingsRoutes) throws Exception {
+        webAdminServer = WebAdminUtils.createWebAdminServer(new DefaultMetricFactory(), domainMappingsRoutes);
+        webAdminServer.configure(NO_CONFIGURATION);
+        webAdminServer.await();
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer)
+                .setBasePath("/domainMappings")
+                .log(LogDetail.METHOD)
+                .build();
+    }
+
+    @BeforeEach
+    void setUp() throws Exception {
+        recipientRewriteTable = spy(new MemoryRecipientRewriteTable());
+
+        createServer(new DomainMappingsRoutes(recipientRewriteTable, new JsonTransformer()));
+    }
+
+    @AfterEach
+    void tearDown() {
+        webAdminServer.destroy();
+    }
+
+    @Nested
+    class NormalBehaviour {
+
+        @Test
+        void addDomainMappingShouldRespondWithNoContent() {
+            // @formatter:off
+            given(with().body("to.com"))
+                        .put("from.com")
+                    .then()
+                        .statusCode(HttpStatus.NO_CONTENT_204)
+                        .body(isEmptyString());
+            // @formatter:on
+        }
+
+        @Test
+        void getDomainMappings() throws RecipientRewriteTableException {
+            Domain expectedDomain = Domain.of("abc.com");
+            MappingSource mappingSource = MappingSource.fromDomain(expectedDomain);
+            ImmutableList<String> expectedAliases = ImmutableList.of("to_1.com", "to_2.com", "to_3.com");
+
+            for (String alias : expectedAliases) {
+                recipientRewriteTable.addAliasDomainMapping(mappingSource, Domain.of(alias));
```

Comment:
We tend to avoid logic (even the simplest one) in our tests, in order to make them easier to read.

Here we would prefer:

```
String alias1 = "to_1.com";
String alias2 = "to_2.com";
String alias3 = "to_3.com";
recipientRewriteTable.addAliasDomainMapping(mappingSource, Domain.of(alias1));
recipientRewriteTable.addAliasDomainMapping(mappingSource, Domain.of(alias2));
recipientRewriteTable.addAliasDomainMapping(mappingSource, Domain.of(alias3));

//...

assertThat(map)
              .containsOnly(entry(expectedDomain.name(), ImmutableList.of(alias1, alias2, alias3)));
```